lib/rules/require-sorted-object-keys.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Requires sorted object keys.
+ *
+ * Types: `Boolean` or `String`
+ *
+ * Values:
+ *  - `true` (alias to `asc`)
+ *  - `"asc"`: requires sorting in ascending order
+ *  - `"desc"`: requires sorting in descending order
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireSortedObjectKeys": "asc"
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * var x = {
+ *  x: 'foo',
+ *  y: 'bar'
+ * }
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var x = {
+ *  y: 'foo',
+ *  x: 'bar'
+ * }
+ * ```
+ */
+
+function asc(a, b) {
+    if (/[_]/.test(a)) {
```

Comment:
This can return false positive for snake cased keys like `snake_case`.
E.g. this object will catch that bug:
```
{
  b_: 'asd',
  a: 'qwe'
}
```